src/cache/cache_las.c

```diff
@@ -682,10 +683,28 @@ __wt_las_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_PAGE *page, WT_MU
 err:
     /* Resolve the transaction. */
     if (local_txn) {
-        if (ret == 0)
+        if (ret == 0) {
             ret = __wt_txn_commit(session, NULL);
-        else
+            txn_rollbacked = ret != 0;
+        } else {
             WT_TRET(__wt_txn_rollback(session, NULL));
+            txn_rollbacked = true;
+        }
+
+        if (txn_rollbacked && multi->supd_entries > 0) {
+            /* We only need to clear the flag on the updates up to where the error occurs. */
+            err_pos = WT_MIN(multi->supd_entries - 1, i);
+
+            /* Traverse the keys again to clear the flags on updates inserted to lookaside. */
+            for (i = 0, list = multi->supd; i <= err_pos; ++i, ++list) {
+                if (list->onpage_upd == NULL)
+                    continue;
+
+                for (upd = list->onpage_upd; upd != NULL; upd = upd->next)
```

Comment:
Since we had a conversation about what the right thing to do is here, it deserves a comment. Something like "Clear the flag on all updates in the chain regardless of whether we wrote the entry to lookaside. That means we might clear the flag for updates that are in lookaside, but that's safe and getting this correct is not worth the complexity of tracking which specific updates were written in the transaction."

Reply:
fixed